signup/signup.js

```diff
@@ -0,0 +1,139 @@
+const inputName = document.querySelector('#nickname');
+const inputPasswordRepeat = document.querySelector('#password-repeat');
+const signupBtn = document.querySelector('.signup-btn');
+const noneName = document.querySelector('.none-nickname-value');
+const nonePasswordRepeat = document.querySelector(
+  '.none-password-repeat-value'
+);
+const missMatchPassword = document.querySelector('.miss-match-password');
+const hidePasswordRepeatIcon = document.querySelector('.password-repeat-hide');
+const showPasswordRepeatIcon = document.querySelector('.password-repeat-show');
+
+// 닉네임 유효성 검사 함수
+function noneNameChecker(name) {
+  if (name !== '') {
+    noneName.classList.add('hide');
+    inputName.classList.remove('error');
+  } else {
+    noneName.classList.remove('hide');
+    inputName.classList.add('error');
+  }
+}
+
+// 비밀번호 재확인 유효성 검사 함수
+function isPasswordRepeatValid(password, rePassword) {
+  if (rePassword !== '') {
+    nonePasswordRepeat.classList.add('hide');
+    inputPasswordRepeat.classList.remove('error');
+    missMatchPassword.classList.add('hide');
+
+    if (password !== rePassword) {
+      missMatchPassword.classList.remove('hide');
+      inputPasswordRepeat.classList.add('error');
+      nonePasswordRepeat.classList.add('hide');
+    }
+  } else {
+    nonePasswordRepeat.classList.remove('hide');
+    inputPasswordRepeat.classList.add('error');
+    missMatchPassword.classList.add('hide');
+  }
+}
+
+// 재확인 비밀번호 표시 함수
+function showPasswordRepeat() {
+  if (inputPasswordRepeat.type == 'password') {
+    inputPasswordRepeat.type = 'text';
+    showPasswordRepeatIcon.classList.remove('hide');
+    hidePasswordRepeatIcon.classList.add('hide');
+  }
+}
+
+// 재확인 비밀번호 숨기기 함수
+function hidePasswordRepeat() {
+  if (inputPasswordRepeat.type == 'text') {
+    inputPasswordRepeat.type = 'password';
+    hidePasswordRepeatIcon.classList.remove('hide');
+    showPasswordRepeatIcon.classList.add('hide');
+  }
+}
+
+// 회원가입 버튼 활성화 함수
+function signupBtnActivation(email, password, name, rePassword) {
+  if (
+    email &&
+    password &&
+    name &&
+    rePassword &&
+    emailFormat.test(email) &&
+    password.length >= 8 &&
+    password === rePassword
+  ) {
```

Comment:
@Seo-Yurim 지금은 괜찮지만, 혹시 조건이 더 많아지게 되면 아래와 같이 해도 괜찮아요! 

```suggestion
const isEmailValid = email && emailFormat.test(email)
const isPasswordValid = password && password.length >= 8
const isRePasswordValid = rePassword && password === rePassword
  if (
    name &&
    isEmailValid &&
    isPasswordValid &&
    isRePasswordValid
  ) {
```

Reply:
오 !! 이 방법이 저는 눈에 더 잘 들어오고 좋은 것 같습니다 ! 말씀해주신 부분이랑 login.js 에서도 버튼 활성화 함수 부분에 적용시켜보았습니다 !